src/runtime/Memory.h

```diff
@@ -90,11 +90,8 @@ class Memory : public Extern {
     void load(ExecutionState& state, uint32_t offset, T* out) const
     {
         checkAccess(state, offset, sizeof(T));
-#if defined(WALRUS_BIG_ENDIAN)
-        *out = *(reinterpret_cast<T*>(&m_buffer[m_sizeInByte - sizeof(T) - offset]));
-#else
-        *out = *(reinterpret_cast<T*>(&m_buffer[offset]));
-#endif
+
+        load(state, offset, 0, out);
```

Comment:
Maybe we could use memcpy since it is optimized by the compiler, but also tells that the data is unaligned. The compiler used an `ldrp` instruction for 64 bit load which does not support unaligned address. It should use two `ldr` instead.

Reply:
I looked at the disassembly in release mode between the two versions, it looks like every instance of `endiannessAwareMemcpy` gets inlined.  Didn't see a single reference to `libc`'s `memcpy`.

Reply:
I would still use memcpy directly here.